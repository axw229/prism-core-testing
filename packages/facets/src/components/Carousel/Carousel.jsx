// @flow
import React, { useContext } from 'react'
import { useIntl } from 'react-intl'
import { Carousel as ToolkitCarousel } from '@prism/toolkit'
import ConfigurationContext, {
  type ConfigurationContextType
} from '../../contexts/ConfigurationContext/ConfigurationContext'
import Iconography from '../Iconography/Iconography'
import { withPrism } from '../ToolkitComponents'

const Carousel = (props) => {
  const { formatMessage } = useIntl()
  const { cvw = {} } = useContext<ConfigurationContextType>(ConfigurationContext)
  const { carouselBtn = {} } = cvw

  const leftButton = {
    icon: carouselBtn.iconLeft && <Iconography name={carouselBtn?.iconLeft} style={{ height: '.95rem' }} />,
    label: formatMessage({ id: 'PREVIOUS' })
  }
  const rightButton = {
    icon: carouselBtn.iconRight && <Iconography name={carouselBtn?.iconRight} style={{ height: '.95rem' }} />,
    label: formatMessage({ id: 'NEXT' })
  }

  return <ToolkitCarousel leftButton={leftButton} rightButton={rightButton} {...props} />
}

export default withPrism(Carousel)
