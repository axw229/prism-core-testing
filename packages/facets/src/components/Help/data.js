// @flow
import { shouldAllowFeature } from '../../shared/utils/featureSwitch.util'
import { FEATURE_EXCLUSIONS } from '../../constants/configurations'

export const helpTabs = [{
  'id': 0,
  'name': 'ICONS_&_BUTTONS',
  'header': 'HELPFUL_HINTS.HEADER.ICONS_&_BUTTONS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ICONS_&_BUTTONS',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'plus-circle', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.ADD_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.ADD_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'folder', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.SAVE']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'trash', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.TRASH_A_COLOR',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.TRASH_A_COLOR']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'clone', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.MORE_SCENES',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.MORE_SCENES']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'align-justify', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.COLOR_DETAILS',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.COLOR_DETAILS']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'signal-3', 'rotate': 235, 'size': 'lg' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.GRAB_&_REORDER',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_CONTENT.GRAB_&_REORDER']
  }, {
    'fontAwesomeIcon': {},
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.PAINT_SCENE',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 1,
  'name': 'MY_COLOR_PALETTE',
  'header': 'HELPFUL_HINTS.HEADER.MY_COLOR_PALETTE',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.MY_COLOR_PALETTE',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'myColorPaletteBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPalette3'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.TAP_AND_HOLD_THE_BOTTOM_EDGE_OF_A_COLOR_SWATCH_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette4'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPalette5'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.MY_COLOR_PALETTE',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.MY_COLOR_PALETTE'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_COLOR_DETAILS_ICON_TRASH_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.AN_ACTIVE_COLOR_STATE',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.MY_COLOR_PALETTE.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'myColorPaletteMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 2,
  'name': 'ADDING_COLORS',
  'header': 'HELPFUL_HINTS.HEADER.ADDING_COLORS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.ADDING_COLORS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'addColorBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColor1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColor2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColor3'
  }],
  'subContent': [{
    'header': 'HELPFUL_HINTS.SUB_CONTENT_HEADER.ADDING_COLORS',
    'content': 'HELPFUL_HINTS.SUB_CONTENT_DESCRIPTION.ADDING_COLORS'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_ADD_A_COLOR_ICON_IN_A_COLOR_TILE_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.CLICK_THE_ADD_COLOR_ICON_TO',
    'role': 'text',
    'imagePathKey': 'addColorMobile2'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.ADDING_COLORS.START_A_NEW_PALETTE_TO_ADD_MORE_THAN_EIGHT_COLORS',
    'role': 'text',
    'imagePathKey': 'addColorMobile3'
  }],
  'isHiddenMobile': false
}, {
  'id': 3,
  'name': 'COLOR_DETAILS',
  'header': 'HELPFUL_HINTS.HEADER.COLOR_DETAILS',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.COLOR_DETAILS',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'colorDetailBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetail1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetail2'
  }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.CLICK_THE_COLOR_DETAILS_ICON_TO',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.COLOR_DETAILS.PREVIEW_A_COLOR_IN_COLOR_DETAILS',
    'role': 'text',
    'imagePathKey': 'colorDetailMobile2'
  }],
  'isHiddenMobile': false
}, {
  'id': 4,
  'name': 'PAINTING_MY_OWN_PHOTO',
  'header': 'HELPFUL_HINTS.HEADER.PAINTING_MY_OWN_PHOTO',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.PAINTING_MY_OWN_PHOTO',
  'content': [{
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'fill-drip', 'rotate': 0, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINT_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINT_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'brush', 'rotate': 45 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.PAINTBRUSH',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.PAINTBRUSH']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'mouse-pointer', 'rotate': -20, 'flip': 'horizontal' },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.SELECT',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.SELECT']
  }, {
    'fontAwesomeIcon': { 'variant': 'fa', 'icon': 'eraser', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ERASE',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ERASE']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'plus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DEFINE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.DEFINE_AREA']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'draw-polygon', 'rotate': 10 }, {
      'variant': 'fal',
      'icon': 'minus',
      'rotate': 0,
      'size': 'xs'
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REMOVE_AREA',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.REMOVE_AREA']
  }, {
    'fontAwesomeIcon': { 'variant': 'fal', 'icon': 'search-plus', 'rotate': 0 },
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.ZOOM',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.ZOOM']
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'undo-alt', 'rotate': 0 }, {
      'variant': 'fa',
      'icon': 'redo-alt',
      'rotate': 0
    }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNDO_&_REDO',
    'iconInfoContent': ['HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_CONTENT.UNDO_&_REDO'],
    'isUndoRedo': true
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'redo-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.REDO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fa', 'icon': 'eye', 'flip': 'horizontal' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.HIDE_PAINT',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'info-circle' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.INFO',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'trash-alt' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.DELETE_GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-group' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.GROUP',
    'iconInfoContent': []
  }, {
    'fontAwesomeIcon': [{ 'variant': 'fal', 'icon': 'object-ungroup' }],
    'iconInfoName': 'HELPFUL_HINTS.CONTENT.PAINTING_MY_OWN_PHOTO.ICON_INFO_NAME.UNGROUP',
    'iconInfoContent': []
  }],
  'imageList': '',
  'isHiddenMobile': false
}, {
  'id': 5,
  'name': 'SAVING_MY_WORK',
  'header': 'HELPFUL_HINTS.HEADER.SAVING_MY_WORK',
  'subHeader': 'HELPFUL_HINTS.SUB_HEADER.SAVING_MY_WORK',
  'content': '',
  'imageList': [{
    'alt': '',
    'role': '',
    'imagePathKey': 'savingBg'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'saving1'
  }, { 'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO', 'role': 'text', 'imagePathKey': 'saving3' }],
  'imageListMobile': [{
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.CLICK_SAVE_ICON_TO',
    'role': 'text',
    'imagePathKey': 'savingMobile1'
  }, {
    'alt': 'HELPFUL_HINTS.IMAGE_ALT.SAVING_MY_WORK.NAME_YOUR_CURRENT_PHOTO',
    'role': 'text',
    'imagePathKey': 'savingMobile2'
  }],
  'isHiddenMobile': false
}]

export const helpHeader = 'HELPFUL_HINTS.TITLE'

export const filterHelpItems = (contents: string[], contentsHiddenMobile: string[], featureExclusions: string[]): typeof helpTabs => {
  // Filtering Help Section details:
  // Icons & Buttons - Save hint
  !shouldAllowFeature(featureExclusions, FEATURE_EXCLUSIONS.documentSaving) && helpTabs[0].content.forEach((icon, index) => {
    icon.iconInfoName === 'HELPFUL_HINTS.CONTENT.ICONS_&_BUTTONS.ICON_INFO_NAME.SAVE' && helpTabs[0].content.splice(index, 1)
  })

  const helpTabNames = helpTabs.map(item => item.name)
  contentsHiddenMobile.forEach(content => { helpTabs[helpTabNames.indexOf(content)].isHiddenMobile = true })

  return helpTabs.filter(item => contents.indexOf(item.name) !== -1)
}
