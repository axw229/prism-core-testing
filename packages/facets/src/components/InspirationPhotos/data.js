// @flow
import tinycolor from '@ctrl/tinycolor'
import { type Color } from 'src/shared/types/Colors.js.flow'
import uniqueId from 'lodash/uniqueId'

export const throttleDragTime = 5
// we need this two const to make sure cursor always point to the center of preview circle
export const activedPinsHalfWidth = 24

export const cloneColorPinsArr = (pins) => {
  return pins.map(pin => ({ ...pin }))
}

export const getRGBInitPins = (acgColors: Array<Object>) => {
  return acgColors.map<Object>((acgColor: any): Object => {
    return {
      rgbArray: [acgColor.r, acgColor.g, acgColor.b],
      rgbValue: `rgb(${acgColor.r},${acgColor.g},${acgColor.b})`,
      translateValueX: acgColor.x,
      translateValueY: acgColor.y
    }
  })
}

export const rgb2lab = (rgb: Array<any>) => {
  let r = rgb[0] / 255
  let g = rgb[1] / 255
  let b = rgb[2] / 255
  let x; let y; let z

  r = (r > 0.04045) ? Math.pow((r + 0.055) / 1.055, 2.4) : r / 12.92
  g = (g > 0.04045) ? Math.pow((g + 0.055) / 1.055, 2.4) : g / 12.92
  b = (b > 0.04045) ? Math.pow((b + 0.055) / 1.055, 2.4) : b / 12.92

  x = (r * 0.4124 + g * 0.3576 + b * 0.1805) / 0.95047
  y = (r * 0.2126 + g * 0.7152 + b * 0.0722) / 1.00000
  z = (r * 0.0193 + g * 0.1192 + b * 0.9505) / 1.08883

  x = (x > 0.008856) ? Math.pow(x, 1 / 3) : (7.787 * x) + 16 / 116
  y = (y > 0.008856) ? Math.pow(y, 1 / 3) : (7.787 * y) + 16 / 116
  z = (z > 0.008856) ? Math.pow(z, 1 / 3) : (7.787 * z) + 16 / 116

  return [(116 * y) - 16, 500 * (x - y), 200 * (y - z)]
}

export type RenderingPin = {
  isActiveFlag: boolean,
  name: string,
  colorNumber: number,
  rgbValue: string,
  pinNumber: number,
  isContentLeft: boolean,
  translateX: number,
  translateValueY: number,
  pinId: string
}

export const renderingPins = (initPins: Array<Object>, canvasOffsetWidth: number, canvasOffsetHeight: number, colors: Color[]): RenderingPin[] => {
  return getRGBInitPins(initPins).map<Object>((acgColor: any, index: number): Object => {
    const color: Color = findClosestColor(acgColor.rgbArray, colors)
    const calculateTranslateX = acgColor.translateValueX * canvasOffsetWidth
    const calculateTranslateY = acgColor.translateValueY * canvasOffsetHeight

    return {
      ...color,
      isActiveFlag: (initPins.length - 1 === index),
      rgbValue: `rgb(${color.red},${color.green},${color.blue})`,
      pinNumber: index,
      pinId: uniqueId('pin_'),
      isContentLeft: calculateTranslateX < canvasOffsetWidth / 2,
      translateX: Math.min(canvasOffsetWidth - 2 * activedPinsHalfWidth, calculateTranslateX),
      translateY: Math.min(canvasOffsetHeight - 2 * activedPinsHalfWidth, calculateTranslateY)
    }
  })
}

export const findClosestColor = ([red, green, blue]: [number, number, number], colors: Color[]): Color => {
  const currentPixelHex: string = tinycolor(`rgb (${red}, ${green}, ${blue})`).toHex()

  return colors.reduce(([closestColor, smallestDistanceSoFar], color): [Color, number] => {
    const colorDistance: number = measureDistance(currentPixelHex, tinycolor(`rgb(${color.red},${color.green},${color.blue})`).toHex())
    return colorDistance < smallestDistanceSoFar ? [color, colorDistance] : [closestColor, smallestDistanceSoFar]
  }, [colors[0], 99999])[0]
}

export const measureDistance = (color1: string, color2: string): number => {
  const position1 = getChromaLuma(color1)
  const position2 = getChromaLuma(color2)
  const redDelta = position1[0] - position2[0]
  const greenDelta = position1[1] - position2[1]
  const lumaDelta = position1[2] - position2[2]
  const distance = redDelta * redDelta + greenDelta * greenDelta + lumaDelta * lumaDelta
  return distance
}

export const getChromaLuma = (rgbString: string) => {
  const color = tinycolor(rgbString)
  const luminosity = color.getLuminance()
  const colorRgb = color.toRgb()
  const red = colorRgb.r
  const green = colorRgb.g
  const blue = colorRgb.b
  const sum = red + green + blue
  let redChroma
  let greenChroma
  if (sum > 25) {
    redChroma = red / sum
    greenChroma = green / sum
  } else {
    redChroma = 1 / 3
    greenChroma = 1 / 3
  }

  return [redChroma, greenChroma, luminosity]
}
