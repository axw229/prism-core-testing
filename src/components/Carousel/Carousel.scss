.prism-slick-carousel {
  &__wrapper {
   position: relative;
   width:100%;
    &__content {
      display: flex;
      justify-content: space-between;
      &__prev-btn__wrapper{
        align-items: center;
        display: flex;
        position: absolute;
        left: 0;
        top: 50%;
        transform: translate(0%, -50%);
        z-index:1000;
      }
      &__next-btn__wrapper{
        align-items: center;
        display: flex;
        position: absolute;
        right: 0;
        top: 50%;
        transform: translate(0%, -50%);
        z-index:1000;
      }
      &__buttons {
        border: 1px solid #f2f2f2;
        cursor: pointer;
        font-size: 1em;
        outline: none;
        padding: 1em;
        visibility: visible;
        &--visible {
          visibility: hidden;
        }
        &:hover {
          border: 1px solid #2cabe2;
        }
      }
      &__list__wrapper {
        height: 100%;
        padding: 3em 0;
        width: 100%;
        overflow: hidden;
      }
    }
    &__indicators {
      bottom: 0;
      display: flex;
      justify-content: center;
      margin: 10px auto;
      position: absolute;
      width: 100%;
      &__icons {
        margin: 0 5px;
        &--active {
          color: #2cabe2;
        }
        &--unactive {
          color: #c8c8c8;
        }
      }
    }
  }
}

.collection-list {
  &__wrapper {
    display: inline-flex;
    flex-wrap: wrap;
    justify-content: center;
    width: 100%;
  }
  &__container{
    white-space: nowrap;
    transition-duration: 1000ms;
    transition-timing-function: ease-in-out;
  }
}