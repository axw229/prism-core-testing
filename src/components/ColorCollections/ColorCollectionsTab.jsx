// @flow
import React, { useState } from 'react'
import './ColorCollections.scss'

type Props = {
  collectionTabs: Array,
  showTab: Function,
  tabIdShow: string
}

const baseClass = 'color-collections'
export const tabListSelect = `${baseClass}__tab-list-select`
export const tabListHeading = `${baseClass}__tab-list-heading`
export const tabListDropdownMobile = `${baseClass}__tab-list-dropdown-mobile`
export const tabList = `${baseClass}__tab-list`
export const tabListActive = `${baseClass}__tab-list--active`
export const tabListInactive = `${baseClass}__tab-list--inactive`
export const tabListItem = `${baseClass}__tab-list-item`
export const tabListItemActive = `${baseClass}__tab-list-item--active`

function ColorCollectionsTab (props: Props) {
  const { collectionTabs, showTab, tabIdShow } = props
  const [tabListMobileShow, showTabListMobile] = useState(false)
  const tabActive = collectionTabs.find(tab => tab.id === tabIdShow).tabName
  const tabShowName = (tabActive !== undefined) ? tabActive : 'Choose collection'

  return (
    <div className={`${tabListSelect}`}>
      <span className={`${tabListHeading}`}>Choose a Collection</span>
      <span className={`${tabListDropdownMobile}`} tabIndex='-1' role='button' onKeyDown={() => {}} onClick={() => showTabListMobile(!tabListMobileShow)}>{tabShowName}</span>
      <ul className={`${tabList} ${(tabListMobileShow) ? `${tabListActive}` : `${tabListInactive}`}`}>
        {collectionTabs.map((tab, id) => {
          return (
            <li data-testid={`${tab.id}`} role='presentation' onKeyDown={() => {}} className={`${tabListItem} ${(tab.id === tabIdShow) ? `${tabListItemActive}` : ``}`} key={tab.id} onClick={() => {
              if (tab.id !== tabIdShow) {
                showTab(tab.id)
              }
              showTabListMobile(!tabListMobileShow)
            }}>
              {tab.tabName}
            </li>
          )
        })}
      </ul>
    </div>
  )
}

export default ColorCollectionsTab
