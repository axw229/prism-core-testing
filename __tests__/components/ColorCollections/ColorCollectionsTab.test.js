/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
// import { renderHook, act } from 'react-hooks-testing-library'
import { render, fireEvent, getByTestId } from '@testing-library/react'
import ColorCollectionsTab, {
  tabListSelect,
  tabList, tabListItem,
  tabListDropdownMobile,
  tabListActive,
  tabListInactive
} from 'src/components/ColorCollections/ColorCollectionsTab'
import { collectionTabs } from 'src/components/Carousel/data'

let defaultProps = {
  collectionTabs: collectionTabs,
  showTab: jest.fn(),
  tabIdShow: 'tab1'
}

const getColorCollectionsTab = (props) => {
  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<ColorCollectionsTab {...newProps} />)
}

describe('ColorCollectionsTab', () => {
  let colorCollectionsTab
  beforeAll(() => {
    if (!colorCollectionsTab) {
      colorCollectionsTab = getColorCollectionsTab()
    }
  })

  it('should match snapshot', () => {
    expect(colorCollectionsTab).toMatchSnapshot()
  })

  it('should render tabListSelect class div', () => {
    expect(colorCollectionsTab.find(`div.${tabListSelect}`).exists()).toBe(true)
  })

  it('should render tabListDropdownMobile class span', () => {
    expect(colorCollectionsTab.find(`span.${tabListDropdownMobile}`).exists()).toBe(true)
  })

  it('should render tabList class ul', () => {
    expect(colorCollectionsTab.find(`ul.${tabList}`).exists()).toBe(true)
  })

  it('should render number of li with class tabListItem equal to items in collectionTabs', () => {
    expect(colorCollectionsTab.find(`li.${tabListItem}`)).toHaveLength(collectionTabs.length)
  })

  it('should render li content equal to each tabs tabName in collectionTabs', () => {
    collectionTabs.map((tab, index) => {
      expect(colorCollectionsTab.find(`li.${tabListItem}`).contains(tab.tabName)).toBe(true)
    })
  })

  it('should change ul class when span with class tabListDropdownMobile is clicked', () => {
    const tabListUl = colorCollectionsTab.find(`ul.${tabListInactive}`)
    colorCollectionsTab.find(`span.${tabListDropdownMobile}`).simulate('click')
    if (tabListUl) {
      expect(colorCollectionsTab.find(`ul.${tabListActive}`).exists()).toBe(true)
    } else {
      expect(colorCollectionsTab.find(`ul.${tabListInactive}`).exists()).toBe(true)
    }
  })
})

describe('ColorCollectionsTab with react-testing-library', () => {
  let colorCollectionsTab
  beforeAll(() => {
    if (!colorCollectionsTab) {
      colorCollectionsTab = render(<ColorCollectionsTab {...defaultProps} />)
    }
  })

  it('should call showTab when inactive tab is clicked', () => {
    const { container } = colorCollectionsTab
    collectionTabs.map((tab, index) => {
      const tabLi = getByTestId(container, tab.id)
      fireEvent.click(tabLi)
      if (tab.id !== defaultProps.tabIdShow) {
        expect(defaultProps.showTab).toHaveBeenCalled()
      } else {
        expect(defaultProps.showTab).not.toHaveBeenCalled()
      }
    })
  })
})
