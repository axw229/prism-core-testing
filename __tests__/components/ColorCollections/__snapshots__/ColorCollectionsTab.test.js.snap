// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCollectionsTab should match snapshot 1`] = `
<div
  className="color-collections__tab-list-select"
>
  <span
    className="color-collections__tab-list-heading"
  >
    Choose a Collection
  </span>
  <span
    className="color-collections__tab-list-dropdown-mobile"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="-1"
  >
    Most Popular
  </span>
  <ul
    className="color-collections__tab-list color-collections__tab-list--inactive"
  >
    <li
      className="color-collections__tab-list-item color-collections__tab-list-item--active"
      data-testid="tab1"
      key="tab1"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Most Popular
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab2"
      key="tab2"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Color ID
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab3"
      key="tab3"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Our Finest Whites
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab4"
      key="tab4"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Color Forecast
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab5"
      key="tab5"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Pottery Barn
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab6"
      key="tab6"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      West Elm
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab7"
      key="tab7"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Lifestyle
    </li>
    <li
      className="color-collections__tab-list-item "
      data-testid="tab8"
      key="tab8"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      Kids' Colors
    </li>
  </ul>
</div>
`;
